src/tools/extract-plugin.js

```diff
@@ -191,14 +200,28 @@ const processPluralTranslateString = (cfg, path, state, funcName, types) => {
     msgid_plural,
     msgctxt,
     reference: getLocation(cfg, path, state),
+    extracted: comment,
   };
 };
 
+function getPrecedingComment(line, comments) {
+  return comments.find(comment => comment.line === line - 1)?.comment;
+}
+
 const makeI18nPlugin = cfg => {
   const entries = [];
+  let comments = [];
   const i18nPlugin = ({types}) => {
     return {
       visitor: {
+        Program(path) {
+          comments = path.container.comments
```

Comment:
Would this let us easily use line-before comments in jsx as well instead of having to use the prop?

Reply:
I think so, for example here we do get the comment:

```jsx
<div>
  {/* i18n: comment */}
  <Translate>foo</Translate>
</div>
```